### PR TITLE
[BLO-664] add zod to validate aspect api returns

### PR DESCRIPTION
**Pull request by diegodelrieu**
### Issue / feature description

`Sanitize Aspect API responses with zod`

Some important changes: 
- Added MSW as the mocking library as is recommended by the vitest documentation https://vitest.dev/guide/mocking.html#requests 
- This means that we currently have two mocking libraries, MSW and vi-fetch, the latter is a dead library with 40 stars and no activity in the past year. MSW has over 11k stars and is in active dev. I[ have created a ticket](https://argent.atlassian.net/browse/BLO-734?atlOrigin=eyJpIjoiOTc2ZjA1NTdlNjdiNDJmMWE3OGMwZWQwNThiNzZkZTMiLCJwIjoiaiJ9) to phase out vi-fetch and replace it with MSW. Luckily, there's only a few instances of it.

### Changes

`Add zod library and implemented for Aspect API`

### CODE REVIEW

File: packages/extension/src/ui/features/accountNfts/aspect.service.ts [hunk 1]
```diff
@@ -96,7 +90,7 @@ export const fetchAspectCollection = async (
 
   const url = join(baseUrl, `?${params}`, "&limit=50")
   const assets = await fetchNextAspectCollection(url, account, contractAddress)
-  if (Array.isArray(assets) && assets.length > 0) {
+  if (assets.length > 0) {
```
Comment (simonheys) [inline packages/extension/src/ui/features/accountNfts/aspect.service.ts:93]: Is this guaranteed to be an array now? If not we should keep the check as sometimes the API may return unexpected stuffs

Comment (diegodelrieu) [reply packages/extension/src/ui/features/accountNfts/aspect.service.ts:93]: yes it is zod will throw if it's not an array, but mb this is too strict ?

File: packages/extension/package.json [hunk 2]
```diff
@@ -96,6 +97,7 @@
     "dexie": "^3.2.2",
     "dexie-react-hooks": "^1.1.1",
     "ethers": "^5.5.1",
+    "global": "^4.4.0",
```
Comment (janek26) [inline packages/extension/package.json:100]: can we remove this?

**Review by janek26 — APPROVED** [review]

File: packages/extension/package.json (hunk 2 above)
Comment (diegodelrieu) [reply packages/extension/package.json:100]: ah yes my mistake

**Review by simonheys — APPROVED** [review]

**Pull request merged**